coalib/parsing/filters/CanDetectFilter.py

```diff
@@ -0,0 +1,21 @@
+from coalib.parsing.filters.FilterBase import FilterBase
+
+
+class CanDetectFilter(FilterBase):
+
+    def filter_section_bears(self, bears, args):
+        """
+        Filters the bears by can_detect.
+
+        :param bears:       The dictionary of the sections as keys and list of
+                            bears as values.
+        :param args:        detects that bears are being filtered on.
+        :return:            New dictionary with filtered out bears that matches
+                            any detect from detects.
+        """
+        detects = {detect.lower() for detect in args}
+        return {section:
+                tuple(bear for bear in bears[section]
+                      if {detect.lower()
+                          for detect in bear.CAN_DETECT} & detects)
+                for section in bears.keys()}
```

Comment:
ugh ... can we abstract the dict-building away into the `FilterBase` so we just we have to filter pure bears and don't need to build the dict? I think we are likely duplicating code for new filters with this^^
```python
def filter_bears(self, bears, args):
        local_bears, global_bears = bears
        filtered_local_bears = {section: self.filter_section_bears(bears) for section, bears in local_bears)
        filtered_global_bears = {section: self.filter_section_bears(bears) for section, bears in global_bears)
        return filtered_local_bears, filtered_global_bears
```